### ISSUE TITLE
Wrong interface information for Junos virtual-switch w/ bridge domains using get_network_instances 

### ISSUE THREAD

**Issue by ferfemar** [napalm-automation/napalm#1028]
### Description of Issue/Question

Output of get_network_instances shows no interfaces in Junos routing instance type virtual-switch when interfaces are assigned under a bridge-domain. It counts them as being part of the master/default instance which is not correct.

### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
(Place an ``x`` between the square brackets where applicable)

- [x ] Yes
- [ ] No

### napalm version
(Paste verbatim output from `pip freeze | grep napalm` between quotes below)

```
ferfecky@lablinux:~/vrf-report (napalm-getters)*$ pip freeze | grep napalm
napalm==2.4.0
```

### Network operating system version
(Paste verbatim output from `show version` - or equivalent - between quotes below)

```
root@vMX1> show version
Hostname: vMX1
Model: vmx
Junos: 14.1R1.10
JUNOS Base OS Software Suite [14.1R1.10]
JUNOS Base OS boot [14.1R1.10]
JUNOS Crypto Software Suite [14.1R1.10]
JUNOS Online Documentation [14.1R1.10]
JUNOS Kernel Software Suite [14.1R1.10]
JUNOS Packet Forwarding Engine Support (M320) [14.1R1.10]
JUNOS Packet Forwarding Engine Support (M/T/EX Common) [14.1R1.10]
JUNOS Routing Software Suite [14.1R1.10]
JUNOS Runtime Software Suite [14.1R1.10]
JUNOS Services AACL PIC package [14.1R1.10]
JUNOS Services Application Level Gateway [14.1R1.10]
JUNOS Services Application Level Gateway (xlp64) [14.1R1.10]
JUNOS Services Application Level Gateway (xlr64) [14.1R1.10]
JUNOS AppId Services PIC Package [14.1R1.10]
JUNOS Services AppId PIC package (xlr64) [14.1R1.10]
JUNOS Border Gateway Function PIC package [14.1R1.10]
JUNOS Services Captive Portal and Content Delivery PIC package [14.1R1.10]
JUNOS Services HTTP Content Management PIC package [14.1R1.10]
JUNOS Services HTTP Content Management PIC package (xlr64) [14.1R1.10]
JUNOS IDP Services PIC Package [14.1R1.10]
JUNOS Packet Forwarding Engine Trio Simulation Package [14.1R1.10]
JUNOS Services JFLOW PIC package [14.1R1.10]
JUNOS Services JFLOW PIC package (xlp64) [14.1R1.10]
JUNOS Services LL-PDF PIC package [14.1R1.10]
JUNOS MobileNext PIC package [14.1R1.10]
JUNOS MobileNext PIC package (xlr64) [14.1R1.10]
JUNOS Services Mobile Subscriber Service Container package [14.1R1.10]
JUNOS Services Mobile Subscriber Service PIC package (xlr64) [14.1R1.10]
JUNOS Services NAT PIC package [14.1R1.10]
JUNOS Services NAT PIC package (xlp64) [14.1R1.10]
JUNOS Services NAT PIC package (xlr64) [14.1R1.10]
JUNOS Services PTSP PIC package [14.1R1.10]
JUNOS Services RPM PIC package [14.1R1.10]
JUNOS Services RPM PIC package (xlp64) [14.1R1.10]
JUNOS Services Stateful Firewall PIC package [14.1R1.10]
JUNOS Services Stateful Firewall PIC package (xlp64) [14.1R1.10]
JUNOS Services Stateful Firewall PIC package (xlr64) [14.1R1.10]
JUNOS BSG PIC package [14.1R1.10]
JUNOS Services Crypto Base PIC package [14.1R1.10]
JUNOS Services Crypto Base PIC package [14.1R1.10]
JUNOS Services Crypto Base PIC package(xlr64) [14.1R1.10]
JUNOS Services IPSec PIC package [14.1R1.10]
JUNOS Services IPSec PIC package [14.1R1.10]
JUNOS Services IPSec PIC(xlr64) package [14.1R1.10]
JUNOS Services SSL PIC package [14.1R1.10]

root@vMX1>
```

### Steps to Reproduce the Issue

**This is the Junos config (note the ge-0/0/1.1 interface):**

```
root@vMX1> show configuration routing-instances vswitch1
description "A virtual switch instance";
instance-type virtual-switch;
bridge-domains {
    MY-BD {
        domain-type bridge;
        vlan-id none;
        no-local-switching;
        interface ge-0/0/1.1;
        routing-interface irb.1;
    }
}
```

**Show commands correctly include the interfaces as part of the routing instance:**

```
root@vMX1> show route instance vswitch1 detail
vswitch1:
  Description: A virtual switch instance
  Router ID: 0.0.0.0
  Type: vpls              State: Active
  Interfaces:
    ge-0/0/1.1
  Route-distinguisher: 0:0
  Fast-reroute-priority: low

root@vMX1>
```

**NAPALM does not see the interfaces as being part of the instance and adds it to the default one instead.** I think this is due to the fact that NAPALM uses the router configuration instead of the above show command to find out VRF information. The interfaces when configured under a bridge-domain are not present directly under the VRF configuration where NAPALM probably expects them to be.

```
In [1]: from napalm import get_network_driver
   ...:
   ...: driver = get_network_driver('junos')
   ...:
   ...: device = driver('vMX1', 'root', 'Juniper123')
   ...:
   ...: device.open()
   ...:

In [2]: instances = device.get_network_instances()

In [3]: instances['vswitch1']
Out[3]:
{u'interfaces': {u'interface': {}},
 u'name': u'vswitch1',
 u'state': {u'route_distinguisher': u''},
 u'type': u'virtual-switch'}

In [4]:
```

**Comment by ferfemar** [comment]
Also the type is arguably wrong in this case. Junos identifies the instance as **VPLS**, but NAPALM as **virtual-switch**. Should I open a separate issue for it? @ckishimo

**Comment by ckishimo** [comment]
@ferfemar I don't think this can be quickly fixed at this point. The napalm method is getting all the information from the candidate configuration, instead of using the RPC call `get-instance-information` as you do

So because in the candidate configuration, the instance type is reported as `virtual-switch` napalm returns the same instance-type
```
ckishimo@vmx> show configuration routing-instances vswitch1 | match instance-type
instance-type virtual-switch;
```
I believe one would need to change completely the implementation of this method and use the RPC call which is reporting the instance type as `VPLS`
```
ckishimo@vmx> show route instance vswitch1 detail | match Type      
  Type: vpls              State: Active
```
Then some may prefer to have the instance-type configured while others may want the instance-type reported via the RPC call... so probably a separate issue is needed :wink:

**Comment by mirceaulinic** [comment]
@ckishimo fixed this in #1032.
